config/mbed/CMakeLists.txt

```diff
@@ -0,0 +1,279 @@
+#
+#   Copyright (c) 2020 Project CHIP Authors
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+
+#
+#   @file
+#     CMake sub-project defining 'chip' target which represents CHIP library
+#     and other optional libraries like unit tests, built with 'mbed'
+#     platform.
+#     Since CHIP doesn't provide native CMake support, ExternalProject
+#     module is used to build the required artifacts with GN meta-build
+#     system.
+#
+
+include(ExternalProject)
+include(mbed-util.cmake)
+
+# ==============================================================================
+# Declare configuration variables and define constants
+# ==============================================================================
+# C/C++ compiler flags passed to CHIP build system
+list(APPEND CHIP_CFLAGS)
+
+# C compiler flags passed to CHIP build system
+list(APPEND CHIP_CFLAGS_C)
+
+# C++ compiler flags passed to CHIP build system
+list(APPEND CHIP_CFLAGS_CC)
+
+# CHIP libraries that the application should be linked with
+list(APPEND CHIP_LIBRARIES)
+
+# GN meta-build system arguments in the form of 'key1 = value1\nkey2 = value2...' string
+string(APPEND CHIP_GN_ARGS)
+
+# C/C++ compiler flags which should not be forwarded to CHIP
+# build system (e.g. because CHIP configures them on its own)
+set(CHIP_CFLAG_EXCLUDES 
+    "-fno-asynchronous-unwind-tables"
+    "-fno-common"
+    "-fno-defer-pop"
+    "-fno-reorder-functions"
+    "-ffunction-sections"
+    "-fdata-sections"
+    "-g*"
+    "-O*"
+    "-W*"
+)
+
+# ==============================================================================
+# Helper macros
+# ==============================================================================
+
+macro(chip_gn_arg_string ARG STRING)
+    string(APPEND CHIP_GN_ARGS "${ARG} = \"${STRING}\"\n")
+endmacro()
+
+macro(chip_gn_arg_bool ARG BOOLEAN)
+    if (${BOOLEAN})
+        string(APPEND CHIP_GN_ARGS "${ARG} = true\n")
+    else()
+        string(APPEND CHIP_GN_ARGS "${ARG} = false\n")
+    endif()
+endmacro()
+
+macro(chip_gn_arg_flags ARG CFLAGS)
```

Comment:
I can see it's copied from the nrfconnect code, but note that we've lately slightly changed this approach to make sure that GN args are properly escaped. I suggest you verify that in your solution you can pass flags containing `"` to GN without a build failure.

Reply:
Thanks Damian, indeed we borrowed a lot from nrf connect as it uses CMake too. I will update the CMakeLists.txt with the changes present there.

Reply:
We actually escape the in `mbed-util.cmake` in the function `mbed_get_compile_definitions`